src/ros_filter.cpp

```diff
@@ -1898,6 +1901,9 @@ namespace RobotLocalization
   template<typename T>
   void RosFilter<T>::aggregateDiagnostics(diagnostic_updater::DiagnosticStatusWrapper &wrapper)
   {
+    // This mutex prevents aggregateDiagnostics and addDiagnostic from clashing
+    boost::mutex::scoped_lock lock(diagnosticMapsMutex_);
```

Comment:
This actually protects the maps and the error level attributes. I was trying to find another place to put the mutex. Here is probably ok, although I think it can be after `wrapper.clearSummary();` too.